python/dolfinx/fem/forms.py

```diff
@@ -58,9 +63,30 @@ def code(self) -> str:
         """C code strings"""
         return self._code
 
+    # The below member function definitions are necessary for type checking
```

Comment:
I don't follow - the methods look helpful in any case?

Reply:
I can remove the comment. 

In short, mypy (obviously) cannot static check the C++ part of this dynamically constructed class, so the methods to be duplicated and typed in Python.